library/psa_crypto.c

```diff
@@ -5448,7 +5544,7 @@ psa_status_t psa_key_derivation_setup( psa_key_derivation_operation_t *operation
     return( status );
 }
 
-#if defined(MBEDTLS_MD_C)
+#if defined(MBEDTLS_PSA_BUILTIN_ALG_HKDF)
```

Comment:
In the block currently guarded by this pre-processor directive guard only psa_hkdf_input() with "#if defined(MBEDTLS_PSA_BUILTIN_ALG_HKDF)", the other ones with MBEDTLS_MD_C.